From b2cc8b9a491051d064feb1ebc5a083e329b8bdea Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yb@renesas.com>
Date: Wed, 22 Jul 2020 10:54:43 +0700
Subject: [PATCH 246/504] ARM: dts: r8a7742: Add Fine Display Processor (FDP1
 module) node

This patch support Renesas FDP1 kernel module probe.

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 28 ++++++++++++++++++++++++++--
 1 file changed, 26 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index c3850dc..b7f6073 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -171,6 +171,30 @@
 		dma-channels = <15>;
 	};
 
+	fdp0: fdpm@fe940000 {
+		compatible = "renesas,fdp1";
+		reg = <0 0xfe940000 0 0x2400>;
+		interrupts = <0 262 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7742_CLK_FDP1_0>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+	};
+
+	fdp1: fdpm@fe944000 {
+		compatible = "renesas,fdp1";
+		reg = <0 0xfe944000 0 0x2400>;
+		interrupts = <0 263 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7742_CLK_FDP1_1>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+	};
+
+	fdp2: fdpm@fe948000 {
+		compatible = "renesas,fdp1";
+		reg = <0 0xfe948000 0 0x2400>;
+		interrupts = <0 264 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7742_CLK_FDP1_2>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+	};
+
 	gic: interrupt-controller@f1001000 {
 		compatible = "arm,gic-400";
 		#interrupt-cells = <3>;
@@ -1018,8 +1042,8 @@
 			>;
 			clock-output-names =
 				"vcp1", "vcp0", "vpc1", "vpc0", "ssp_dev",
-				"tmu1", "pvrsrvkm", "tddmac", "fdp2", "fdp1",
-				"fdp0", "tmu3", "tmu2", "cmt0", "tmu0",
+				"tmu1", "pvrsrvkm", "tddmac", "fdp1-2", "fdp1-1",
+				"fdp1-0", "tmu3", "tmu2", "cmt0", "tmu0",
 				"vsp1-du1", "vsp1-du0", "vspr", "vsps";
 		};
 		mstp2_clks: mstp2_clks@e6150138 {
-- 
2.7.4

